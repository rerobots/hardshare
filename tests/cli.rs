// SCL <scott@rerobots.net>
// Copyright (C) 2020 rerobots, Inc.

use std::io::prelude::*;

use assert_cmd::Command;
use tempfile::{tempdir, NamedTempFile};


#[test]
fn prints_version() {
    let mut cmd = Command::cargo_bin("hardshare").unwrap();
    let assert = cmd.arg("-V").assert();
    assert
        .stdout(format!("{}\n", env!("CARGO_PKG_VERSION")))
        .success();
}


#[test]
fn prints_help() {
    let mut cmd = Command::cargo_bin("hardshare").unwrap();
    let assert = cmd.arg("help").assert();
    let output = assert.get_output().clone();
    assert.success();
    insta::assert_display_snapshot!(String::from_utf8(output.stdout).unwrap());
}


#[test]
fn config_requires_arg() {
    let mut cmd = Command::cargo_bin("hardshare").unwrap();
    let assert = cmd.arg("config").assert();
    assert.failure().code(1);
}


#[test]
fn add_token_does_not_exist() {
    let ntf = NamedTempFile::new().unwrap();
    let mut cmd = Command::cargo_bin("hardshare").unwrap();
    let assert = cmd
        .arg("config")
        .arg("--add-key")
        .arg(ntf.path().join("notexist"))
        .assert();
    assert.failure().code(1);
}


#[test]
fn list_config_does_not_exist() {
    let tmphome = tempdir().unwrap();
    let mut cmd = Command::cargo_bin("hardshare").unwrap();
    let assert = cmd
        .env("HOME", tmphome.path())
        .arg("config")
        .arg("--local")
        .arg("-l")
        .assert();
    assert.failure().code(1);
}
