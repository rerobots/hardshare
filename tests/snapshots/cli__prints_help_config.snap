---
source: tests/cli.rs
assertion_line: 36
expression: "String::from_utf8(output.stdout).unwrap()"
---
hardshare-config 
Manage local and remote configuration

USAGE:
    hardshare config [FLAGS] [OPTIONS] [ID]

FLAGS:
    -c, --create               If no local configuration is found, then create
                               one
    -h, --help                 Prints help information
        --include-dissolved    Include configuration data of dissolved workspace
                               deployments
    -l, --list                 Lists configuration
        --local                Only show local configuration data
    -p, --prune                delete files in local API tokens directory that
                               are not valid; to get list of files with errors,
                               try `--list`
    -V, --version              Prints version information

OPTIONS:
        --cprovider <CPROVIDER>
            select a container provider: lxd, docker, podman, proxy

        --assign-image <IMG>
            assign image for cprovider to use (advanced option)

        --add-token <FILE>         add new API token

ARGS:
    <ID>    id of workspace deployment for configuration changes (can be
            unique prefix); this argument is not required if there is only 1
            workspace deployment

